.github/workflows/publish.yaml

```diff
@@ -0,0 +1,22 @@
+name: Publish package to the Maven Central Repository
+on:
+  release:
+    types: [created]
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Maven Central Repository
+        uses: actions/setup-java@v3
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+          server-id: ossrh
+          server-username: MAVEN_USERNAME
+          server-password: MAVEN_PASSWORD
+      - name: Publish package
+        run: make release-all
```

Comment:
I think this one is tricky: we shouldn't invoke `release-all` here.
`release-all` creates the git-commit of the tag which should trigger this workflow after it is merged.
sounds like an endless cycle.

If this one should only publish maven resources - then we invoke only the `make release` or `mvn deploy -P release` and remove that option from `make release-all`

@pkliczewski WDYT?

Reply:
this flow is triggered only on a github release , not on a tag push

Reply:
oh, so my bad here.
you are correct and that was explained in the commit message.

Reply:
I hope this all would work :pray: